### ISSUE TITLE
Generate container config from image manifest

### ISSUE THREAD

**Issue by mathstuf** [opencontainers/runtime-tools#669]
It would be nice to be able to generate a `config.json` based on the output of `docker inspect` or `podman inspect`. Images (or containers) already have knowledge of things like exposed ports, commands to run, etc. and it'd be nice to be able to use that as a base for generating a `config.json`.

Example:

```sh
oci-runtime-tool generate --image image.json …
```

**Comment by q384566678** [comment]
I this it's a good ideal, and I'm prepared to do this.
As you said, before running this function we need a json file of the results of the `docker inspect`, and then we run the generate function on this basis to produce the new json file. Is my understanding correct?

**Comment by mathstuf** [comment]
Yes. I suspect there are multiple different formats out there (Docker-specific, OCIv1, probably others from other tools), but I *think* they all have a MIME type key in there somewhere to tell them apart from each other.

**Comment by mathstuf** [comment]
Also nice would be converting a `systemd.nspawn` file into a `config.json`.
